### ISSUE TITLE
Multi object Entities

### ISSUE THREAD

**Issue by JAremko** [openglobus/openglobus#824]
I experimented a bit with the [Instanced Geo Object demo](https://sandbox.openglobus.org/examples/geoObject) to figure out if I can create something similar but with a multi object observer (camera) dummy.  Basically a body on the surface, that only rotates  with yaw and a head that looks into a direction. The problem is that the head has to rotate with a different basis vector (pivot) to avoid translation against the body.  Also the relationship between the pivots of the body and the head has to be constant with the scale (so adding altitude to the head won't work, since it doesn't scale).   

To implement something like this I imagine there need to be a way to load multiply objects from an obj file  or create a 3d object and then make a basis vector hierarchy so that the head has it's origin relative to the body origin.

**Comment by Zemledelec** [comment]
It sounds like you want a complex object with moving parts. Could you draw a picture with the objects and vectors etc, for better view? I know how to improve geoObject instantiating, but I need to know the cases of how to move the parts of such objects.

**Comment by JAremko** [comment]
https://github.com/user-attachments/assets/4568a81e-ee49-4d10-b146-cee5c14f3121

**Comment by JAremko** [comment]
@Zemledelec  it would also be cool to have a helper function `lookAtLonLat` for orienting a child object in the direction of an another  LonLat location.

**Comment by Zemledelec** [comment]
Yes! Just need to develop user friendly API for this. Configure joints etc. e.g. simple editor as in your video.

**Comment by JAremko** [comment]
@Zemledelec  It would be enough to provide a parenting API that allows adding child origins  "slots" and putting into the slots other nodes so that their origin will transform with the parent's one.

**Comment by JAremko** [comment]
For example, I can load an OBJ file with a root model and a bunch of dummy objects (points) so I can extract their coordinates to crate "slots" without actually loading them into the scene (except the root, the cylinder in the scene I provided).   
Then I'll simply use Blender3D(or any other 3D modeling software)  to position everything as I want.

**Comment by Zemledelec** [comment]
1) I'm going to review shaders and replace it with a quaternion math **(done)**
2) Add children geoObgect option with origin position in the parent system **(in progress)**
3) It looks like there will be a problem with child orientation and general rotation, but I'll try to deal with that with a blocking flag or something similar

**Comment by JAremko** [comment]
Thanks

**Comment by Zemledelec** [comment]
Sorry, I had cold. Now, I'm fine, to be continued...

**Comment by Zemledelec** [comment]
In progress...

**Comment by Zemledelec** [comment]
Sorry it take a lot of time, propably @JAremko you forget this issue. It is still in progress...Sorry again. I almost remade rendering engine, and made a lot of stuff...really important engine things, such as great precision rendering, and deferred rendering, wich will helps create shadows, screen space oclusion, deferred lighting etc cool postprocessing effects. Also, I made more or less featured editor for geoobjects, as you showed aabove. After, I finish precision rendering for all entities(polylines, terrain segments, billboards) I will make the initial task. It will be huge release.

**Comment by Zemledelec** [comment]
Almost done...

**Comment by Zemledelec** [comment]
@JAremko Video of the test branch https://x.com/michaelgevlich/status/1883980537452445803?s=46&t=rLN00G3SrfgjG01f7YarcA


…wow 7 moths ago.

**Comment by JAremko** [comment]
Looks cool. I'll send the link to the dude currently working on the map integration 👍

**Comment by Zemledelec** [comment]
I will prepare some examples here https://github.com/openglobus/og-examples/issues/14
